File: flutter_app/lib/services/data_service.dart
import '../models/campaign.dart';
import '../models/fuel.dart';
import '../models/group.dart';
import '../data/campaigns_data.dart' as campaigns_data;
import '../data/fuel_data.dart' as fuel_data;
import '../data/groups_data.dart' as groups_data;
import '../config/app_config.dart';

class DataService {
  // Campaign methods
  static Campaign? getCampaignById(String id) {
    return campaigns_data.getCampaignById(id);
  }

  static Campaign? getCampaignByCode(String code) {
    return campaigns_data.getCampaignByCode(code);
  }

  static List<Campaign> getCampaignsByGroup(String groupId) {
    return campaigns_data.getCampaignsByGroup(groupId);
  }

  static List<Campaign> getAllCampaigns() {
    return campaigns_data.campaigns;
  }

  static List<Campaign> getAvailableCampaigns() {
    final allCampaigns = campaigns_data.campaigns;
    final filter = AppConfig.current.campaignGroupFilter;
    if (filter == null) return allCampaigns;
    return allCampaigns.where((c) => c.groupId == filter).toList();
  }

  // Fuel methods
  static Fuel? getFuelById(String id) {
    return fuel_data.getFuelById(id);
  }

  static Fuel? getFuelByCampaignAndDay(String campaignId, int day) {
    return fuel_data.getFuelByCampaignAndDay(campaignId, day);
  }

  static List<Fuel> getFuelByCampaign(String campaignId) {
    return fuel_data.getFuelByCampaign(campaignId);
  }

  static List<Fuel> getFuelByDate(String date) {
    return fuel_data.getFuelByDate(date);
  }

  // Group methods
  static Group? getGroupById(String id) {
    return groups_data.getGroupById(id);
  }

  static List<Group> getAllGroups() {
    return groups_data.getAllGroups();
  }

  static List<Group> getAvailableGroups() {
    final availableCampaigns = getAvailableCampaigns();
    final campaignGroupIds = availableCampaigns.map((c) => c.groupId).toSet();
    return groups_data.getAllGroups()
        .where((g) => campaignGroupIds.contains(g.id))
        .toList();
  }
}

File: native-app/src/services/DataService.ts
import { Campaign } from '../models/Campaign';
import { Fuel } from '../models/Fuel';
import { Group } from '../models/Group';
import { Language } from '../models/Language';
import { campaigns, getCampaignById, getCampaignByCode, getCampaignsByGroup } from '../data/campaigns';
import { fuel, getFuelById, getFuelByCampaignAndDay, getFuelByCampaign, getFuelByDate } from '../data/fuel';
import { groups, getGroupById } from '../data/groups';
import { languages, getLanguageByCode } from '../data/languages';

export class DataService {
  // Campaign methods
  static getAllCampaigns(): Campaign[] {
    return campaigns;
  }

  static getCampaign(id: string): Campaign | undefined {
    return getCampaignById(id);
  }

  static getCampaignByCode(code: string): Campaign | undefined {
    return getCampaignByCode(code);
  }

  static getCampaignsByGroup(groupId: string): Campaign[] {
    return getCampaignsByGroup(groupId);
  }

  static searchCampaigns(query: string): Campaign[] {
    const lowerQuery = query.toLowerCase();
    return campaigns.filter(c =>
      c.name.toLowerCase().includes(lowerQuery) ||
      c.shortDescription.toLowerCase().includes(lowerQuery)
    );
  }

  static filterCampaignsByLanguage(languageCode: string): Campaign[] {
    return campaigns.filter(c => c.languages.includes(languageCode));
  }

  // Fuel methods
  static getAllFuel(): Fuel[] {
    return fuel;
  }

  static getFuel(id: string): Fuel | undefined {
    return getFuelById(id);
  }

  static getFuelByCampaignAndDay(campaignId: string, day: number): Fuel | undefined {
    return getFuelByCampaignAndDay(campaignId, day);
  }

  static getFuelByCampaign(campaignId: string): Fuel[] {
    return getFuelByCampaign(campaignId);
  }

  static getFuelByDate(date: string): Fuel[] {
    return getFuelByDate(date);
  }

  static getFuelForCampaigns(campaignIds: string[]): Fuel[] {
    return fuel.filter(f => campaignIds.includes(f.campaignId));
  }

  // Group methods
  static getAllGroups(): Group[] {
    return groups;
  }

  static getGroup(id: string): Group | undefined {
    return getGroupById(id);
  }

  // Language methods
  static getAllLanguages(): Language[] {
    return languages;
  }

  static getLanguage(code: string): Language | undefined {
    return getLanguageByCode(code);
  }
}

